functions/minsurf_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package functions
+
+import (
+	"math"
+	"testing"
+
+	"github.com/gonum/diff/fd"
+	"github.com/gonum/floats"
+)
+
+func TestMinimalSurface(t *testing.T) {
```

Comment:
Okay, this is fine. If we add more functions where this is the case (and I imagine we will), we should break this off into a testGradient function.